client/src/routes/Session/hooks/useSubscribeToSession.ts

```diff
@@ -2,16 +2,40 @@ import {useEffect} from 'react';
 import firestore from '@react-native-firebase/firestore';
 import useSessionState from '../state/state';
 import {Session, SessionData} from '../../../../../shared/src/types/Session';
+import useSessions from '../../Sessions/hooks/useSessions';
+import {useNavigation} from '@react-navigation/native';
+import {NativeStackNavigationProp} from '@react-navigation/native-stack';
+import {
+  ModalStackProps,
+  TabNavigatorProps,
+} from '../../../lib/navigation/constants/routes';
 
 const useSubscribeToSession = (sessionId: Session['id']) => {
   const setSessionState = useSessionState(state => state.setState);
 
+  const {navigate} =
+    useNavigation<
+      NativeStackNavigationProp<TabNavigatorProps & ModalStackProps>
+    >();
+
+  const {fetchSessions} = useSessions();
+
   useEffect(() => {
     const doc = firestore().collection('sessions').doc(sessionId);
 
     const unsubscribe = doc.onSnapshot(
-      documentSnapshot =>
-        setSessionState(documentSnapshot.data() as SessionData),
+      documentSnapshot => {
+        const session = documentSnapshot.data() as SessionData;
+
+        if (!documentSnapshot.exists || session.ended) {
```

Comment:
I'm wondering, maybe `leaveMeeting` and resetting the session state is also necessary here 🤔

Reply:
Ok it actually seems to work fine without it 👌

Reply:
but actually it seems to be crashing before going into outro portal, strange  🤔

Reply:
of course, because it is `ended: true` 🤦

Reply:
maybe `ended: true` scenarios should redirect it to outroPortal if existent

Reply:
I think I'll need to remove the `ended` handling from here and move it into the `IntroPortal` and `ChangingRoom`

Reply:
resolved on 85c74b3